### DISCUSSION TITLE
Why am I not able to connect?

### DISCUSSION THREAD

**Discussion by shadowsocksquestions** [shadowsocks/shadowsocks-rust#473] [Q&A] [answered]
So I can connect on Windows and Android, but not on Linux. I have tried with the python & c (qt5) version for Linux too, but couldn't get them to work. I bought some servers from emptyus.com and I am trying a test config, like below:

{
    "server": "usmvip8.helike.me",
    "server_port": 4470,
    "password": "thepassword",
    "method": "aes-256-gcm",
    "local_address": "127.0.0.1",
    "local_port": 54321
}

In the Ubuntu settings, I turned on the proxy, set the socks address to 127.0.0.1 and the port to 54321, however aren't able to connect. I was able to find a few free shadowsocks servers online before, which I got working in the c (qt5) version, so I guess these settings in Ubuntu should be ok.

Does anyone know what the problem is with this setup?

**Edit**: as requested by zonyitoo, here are the logs of the command `sslocal -c config.json` -vvv (note, I broke some onto multiple lines...to hopefully make it easier to read):

```
2021-04-01T18:11:06.237248114+08:00 INFO  [11263:140212888812032] [sslocal] shadowsocks 1.10.2
2021-04-01T18:11:06.237920012+08:00 TRACE [11263:140212888812032] 
[shadowsocks_service::local] Config { server: [ServerConfig { addr: DomainName("usmvip8.helike.me", 4470),
password: "thepassword", method: AES_256_GCM, enc_key: [225, 250, 86, 154, 193, 101, 161, 77, 183, 212, 
166, 176, 89, 76, 69, 227, 29, 251, 125, 185, 166, 197, 208, 40, 205, 197, 69, 220, 177, 101, 209, 137], 
timeout: None, plugin: None, plugin_addr: None, remarks: None, id: None, mode: TcpOnly }], 
local_addr: None, local: [LocalConfig { addr: SocketAddr(127.0.0.1:54321), protocol: Socks, 
mode: TcpOnly, udp_addr: None, forward_addr: None }], dns: System, ipv6_first: false, no_delay: false, 
nofile: None, outbound_fwmark: None, outbound_bind_interface: None, inbound_send_buffer_size: None, 
inbound_recv_buffer_size: None, outbound_send_buffer_size: None, outbound_recv_buffer_size: None, 
manager: None, config_type: Local, udp_timeout: None, udp_max_associations: None, acl: None }
2021-04-01T18:11:06.269571936+08:00 TRACE [11263:140212888812032]
[shadowsocks::dns_resolver::trust_dns_resolver] initializing DNS resolver with system-config ResolverConfig 
{ domain: None, search: [], name_servers: NameServerConfigGroup([NameServerConfig 
{ socket_addr: 127.0.0.53:53, protocol: Udp, tls_dns_name: None, trust_nx_responses: false }, 
NameServerConfig { socket_addr: 127.0.0.53:53, protocol: Tcp, tls_dns_name: None, trust_nx_responses: false }]) }
 opts ResolverOpts { ndots: 1, timeout: 5s, attempts: 2, rotate: false, check_names: true, edns0: false, validate: false,
ip_strategy: Ipv4thenIpv6, cache_size: 32, use_hosts_file: true, positive_min_ttl: None, negative_min_ttl: None,
positive_max_ttl: None, negative_max_ttl: None, num_concurrent_reqs: 2, preserve_intermediates: false }
2021-04-01T18:11:06.277955230+08:00 INFO  [11263:140212888812032] 
[shadowsocks_service::local::socks::server] shadowsocks socks TCP listening on 127.0.0.1:54321
```

**Comment by zonyitoo** [comment, marked as answer]
Could you paste logs that `sslocal` produced to your terminal?

**Reply by shadowsocksquestions** [reply]
Thanks. I've just updated my main post with the logs then :)

**Reply by zonyitoo** [reply]
Add `-vvv` into command line arguments and see if there are anything special in logs.

**Reply by shadowsocksquestions** [reply]
Ok, have done. I don't know exactly how it should look, but nothing is jumping out at me as being obviously wrong?

**Reply by zonyitoo** [reply]
Yes, obviously, your `sslocal` haven't received any requests.

**Reply by zonyitoo** [reply]
You should see plenty lines of log when you initiate connections through `sslocal`.

**Reply by shadowsocksquestions** [reply]
Awesome. I got it working. I was setting the socks address/port in the Ubuntu settings, which I read that "Chrome uses by default"...not realising though that I had a vpn extension in Chrome, which, despite being off, Chrome was using as the connection profile, rather than the one specified in Ubuntu. That explains why there were no logs then...now there is plenty of logs like you mention
